presto-main/src/test/java/io/prestosql/operator/scalar/TestBitwiseFunctions.java

```diff
@@ -90,4 +93,77 @@ public void testBitwiseXor()
+        assertFunction("bitwise_left_shift(INTEGER'-7', 64)", INTEGER, 0);
+        assertFunction("bitwise_left_shift(INTEGER'-2147483648', 0)", INTEGER, -2147483648);
+        assertFunction("bitwise_left_shift(BIGINT'7', 2)", BIGINT, 7L << 2);
+        assertFunction("bitwise_left_shift(BIGINT'-7', 2)", BIGINT, -7L << 2);
+        assertFunction("bitwise_left_shift(BIGINT'-7', 64)", BIGINT, 0L);
+    }
+
+    @Test
+    public void testBitwiseRightShift()
+    {
+        assertFunction("bitwise_right_shift(TINYINT'7', 2)", TINYINT, (byte) (7 >>> 2));
+        assertFunction("bitwise_right_shift(TINYINT'-7', 2)", TINYINT, (byte) 62);
+        assertFunction("bitwise_right_shift(TINYINT'-7', 64)", TINYINT, (byte) 0);
+        assertFunction("bitwise_right_shift(TINYINT'-128', 0)", TINYINT, (byte) -128);
+        assertFunction("bitwise_right_shift(SMALLINT'7', 2)", SMALLINT, (short) (7 >>> 2));
+        assertFunction("bitwise_right_shift(SMALLINT'-7', 2)", SMALLINT, (short) 16382);
+        assertFunction("bitwise_right_shift(SMALLINT'-7', 64)", SMALLINT, (short) 0);
+        assertFunction("bitwise_right_shift(SMALLINT'-32768', 0)", SMALLINT, (short) -32768);
+        assertFunction("bitwise_right_shift(INTEGER'7', 2)", INTEGER, 7 >>> 2);
+        assertFunction("bitwise_right_shift(INTEGER'-7', 2)", INTEGER, 1073741822);
+        assertFunction("bitwise_right_shift(INTEGER'-7', 64)", INTEGER, 0);
+        assertFunction("bitwise_right_shift(INTEGER'-2147483648', 0)", INTEGER, -2147483648);
+        assertFunction("bitwise_right_shift(BIGINT'7', 2)", BIGINT, 7L >>> 2);
+        assertFunction("bitwise_right_shift(BIGINT'-7', 2)", BIGINT, -7L >>> 2);
+        assertFunction("bitwise_right_shift(BIGINT'-7', 64)", BIGINT, 0L);
+    }
+
+    @Test
+    public void testBitwiseRightShiftArithmetic()
+    {
+        assertFunction("bitwise_right_shift_arithmetic(TINYINT'7', 2)", TINYINT, (byte) -63);
+        assertFunction("bitwise_right_shift_arithmetic(TINYINT'-7', 2)", TINYINT, (byte) (-7 >> 2));
+        assertFunction("bitwise_right_shift_arithmetic(TINYINT'7', 64)", TINYINT, (byte) 0);
+        assertFunction("bitwise_right_shift_arithmetic(TINYINT'-7', 64)", TINYINT, (byte) -1);
+        assertFunction("bitwise_right_shift_arithmetic(TINYINT'-128', 0)", TINYINT, (byte) -128);
+        assertFunction("bitwise_right_shift_arithmetic(SMALLINT'7', 2)", SMALLINT, (short) -16383);
+        assertFunction("bitwise_right_shift_arithmetic(SMALLINT'-7', 2)", SMALLINT, (short) (-7 >> 2));
+        assertFunction("bitwise_right_shift_arithmetic(SMALLINT'7', 64)", SMALLINT, (short) 0);
+        assertFunction("bitwise_right_shift_arithmetic(SMALLINT'-7', 64)", SMALLINT, (short) -1);
+        assertFunction("bitwise_right_shift_arithmetic(SMALLINT'-32768', 0)", SMALLINT, (short) -32768);
+        assertFunction("bitwise_right_shift_arithmetic(INTEGER'7', 2)", INTEGER, -1073741823);
```

Comment:
This seems wrong. 7 fits in an INTEGER as a positive number. Shifting to the right should not turn it into a negative number. The issue is that that operation is or-ing the value with ~INTEGER_MASK.